src/nodes.jl

```diff
@@ -95,11 +113,72 @@ This is the most common `DispatchNode`.
 @auto_hash_equals type Op <: DispatchNode
     result::DeferredFuture
     func::Function
+    label::String
     args
     kwargs
 end
 
-Op(func::Function, args...; kwargs...) = Op(DeferredFuture(), func, args, kwargs)
+function Op(func::Function, args...; kwargs...)
+    Op(
+        DeferredFuture(),
+        func,
+        string(Symbol(func)),
+        args,
+        kwargs
+    )
+end
+
+"""
+    summary(op::Op)
+
+Returns a string representation of the `Op`
+with its label and the args/kwargs types.
+
+NOTE: if an arg/kwarg is a `DispatchNode` with a label
+it will be printed with that arg.
+"""
+function Base.summary(op::Op)
+    function val_summary(val)
+        result = split(string(typeof(val)), '.')[end]
+
+        if isa(val, DispatchNode) && has_label(val)
+            label = get_label(val)
+            result = "$result($label)"
+        end
+
+        return result
+    end
+
+    args = map(arg -> val_summary(arg), op.args)
+    kwargs = map(op.kwargs) do kwarg
+        "$(kwarg[1]) => $(val_summary(kwarg[2]))"
+    end
+
+    all_args = join([args..., kwargs...], ',')
+    return "Op($(op.label), $all_args)"
```

Comment:
That seems kind of verbose cause in a lot of cases the op.label will just be the name of op.func.